streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -98,10 +108,17 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
         }
     }
 
+    private void checkForException() {
+        if (sendException != null) {
+            throw new StreamsException(String.format("%s exception caught when producing", logPrefix), sendException);
+        }
+    }
+
     @Override
     public void flush() {
         log.debug("{} Flushing producer", logPrefix);
         this.producer.flush();
+        checkForException();
```

Comment:
I did have it checking before and after flush, but decided to flush and then check. The reason being that we probably want to close the producer, that will also causes the messages to be flushed.